File: web-client/types/dtos/event.ts
import type { IRegion } from '~/types/dtos/region';
import type { IUser } from '~/types/dtos/user';

export interface IEvent {
  readonly id: number;
  readonly name: string;
  readonly region_id?: string;
  readonly discipline: string;
  readonly datetime: Date;
  readonly description?: string;
  readonly is_approved_event?: string;
  readonly member_created_id?: string;
}

export interface IEventDetail {
  readonly event: IEvent;
  readonly region?: IRegion;
  readonly user?: IUser;
}

File: python-service/app/data/domains/event.py
from datetime import datetime
from enum import Enum
from typing import Optional, List

from pydantic import BaseModel


class Event(BaseModel):
    id: Optional[str] = None
    region_id: str
    name: str
    location: str
    participants_count: int
    start_date: datetime
    end_date: datetime
    member_created_id: Optional[str] = None
    discipline: Optional[str] = None
    description: Optional[str] = None
    documents_ids: List[str] = []
    protocols_ids: List[str] = []
    is_approved_event: bool

class EventFilter(str, Enum):
    regions = 'regions'
    disciplines = 'disciplines'
